### PR TITLE
Correct bug in parallel checkpoints

### PR DESCRIPTION
**Pull request by shahabgol**
Correct a bug in parallel checkpoints

### CODE REVIEW

File: include/dem/dem.h [hunk 1]
```diff
@@ -278,6 +278,7 @@ class DEMSolver
   Particles::ParticleHandler<dim, dim> particle_handler;
   unsigned int                         contact_detection_step;
   unsigned int                         load_balance_step;
+  unsigned int                         checkpoint_step;
```
Comment (blaisb) [inline include/dem/dem.h:281]: Why are those variables not bool instead of ints?

File: source/dem/dem.cc [hunk 2]
```diff
@@ -679,6 +680,8 @@ DEMSolver<dim>::solve()
                       particle_handler);
 
       update_moment_of_inertia(particle_handler, MOI);
+
+      checkpoint_step = 1;
```
Comment (blaisb) [inline source/dem/dem.cc:684]: Why not a boolean? then set to true

File: source/dem/dem.cc [hunk 3]
```diff
@@ -722,8 +725,11 @@ DEMSolver<dim>::solve()
 
       // Sort particles in cells
       if (particles_insertion_step || load_balance_step ||
-          contact_detection_step)
+          contact_detection_step || checkpoint_step)
         {
+          // Reset checkpoint step
+          checkpoint_step = 0;
+
```
Comment (blaisb) [inline source/dem/dem.cc:732]: Should be boolean, otherwise good

**Review by blaisb — COMMENTED** [review]
Can you switch type from unsigned int to bool?

File: include/dem/dem.h (hunk 1 above)
Comment (shahabgol) [reply include/dem/dem.h]: Changed to bool.

File: source/dem/dem.cc (hunk 2 above)
Comment (shahabgol) [reply source/dem/dem.cc]: Done.

File: source/dem/dem.cc (hunk 3 above)
Comment (shahabgol) [reply source/dem/dem.cc]: Done.

**Comment by shahabgol** [comment]
> Can you switch type from unsigned int to bool?

Changed to bool.

File: include/dem/dem.h [hunk 4]
```diff
@@ -278,6 +278,7 @@ class DEMSolver
   Particles::ParticleHandler<dim, dim> particle_handler;
   unsigned int                         contact_detection_step;
   unsigned int                         load_balance_step;
```
Comment (blaisb) [inline include/dem/dem.h:280]: Can you change these ones also while you are it :)

**Review by blaisb — COMMENTED** [review]
Can you change the leftover ints while you are at it?
I'll merge afterwards.

File: include/dem/dem.h (hunk 4 above)
Comment (shahabgol) [reply include/dem/dem.h]: done.

**Comment by shahabgol** [comment]
> Can you change the leftover ints while you are at it?
> I'll merge afterwards.

load_balance_step and contact_detection_step types are changed to bool.

**Review by blaisb — APPROVED** [review]

**Pull request merged**